### PR TITLE
feat: added IP Tag Support (#20)

### LINKED ISSUE #20: Automatically set ip tag
There's already a constant for it: `TRACER_IP_TAG_KEY`.

Jaeger [uses](https://github.com/jaegertracing/jaeger/blob/243b56f52973bec79bb641ce2e670c727b34d2c6/model/adjuster/clockskew.go#L74-L96) `ip` tag to group spans around individual machines for clock skew adjustment and with futures it's easy to mess up and make child spans outlive parents, triggering time travel bamboozle:

* https://github.com/jaegertracing/jaeger/pull/2052

It's probably best to infer `ip` tag automatically just like C++ and Go clients do.

### PR DESCRIPTION
**Pull request by anirudhRowjee**
This Adds The IP Tag to the Reporter's tags, and should close #20 . Marked as WIP as I may want to add tests where feasible.

### CODE REVIEW

File: src/reporter.rs [hunk 1]
```diff
@@ -27,7 +28,6 @@ impl JaegerCompactReporter {
     }
 
     /// Sets the address of the report destination agent to `addr`.
-    ///
```
Comment (sile) [inline src/reporter.rs]: Is this an intended change?
If not, please revert the diff.

**Review by sile — CHANGES_REQUESTED** [review]
Thank you for your PR.
This PR almost looks good. I'll merge this PR once you address a minor comment I left.

File: src/reporter.rs (hunk 1 above)
Comment (anirudhRowjee) [reply src/reporter.rs]: my bad, this wasn't intended. I've pushed the change.

**Review by sile — APPROVED** [review]
👍

**Pull request merged**